test/unix/force_merge.ml

```diff
@@ -349,6 +349,38 @@ let test_is_merging () =
   add_binding_and_merge ~hook:(after (f "after" true));
   add_binding_and_merge ~hook:(after_clear (f "after clear" true))
 
+let test_non_blocking_clear () =
+  let* Context.{ rw; _ } = Context.with_empty_index () in
+  let add_bindings () =
+    let k1, v1 = (Key.v (), Value.v ()) in
+    Index.replace rw k1 v1
+  in
+  let lock () =
+    let m = Mutex.create () in
+    Mutex.lock m;
+    m
+  in
+  let merge_started = lock () and merge_ended = lock () and merge = lock () in
+  let merge_hook =
+    Hook.v @@ function
+    | `Before ->
+        Mutex.unlock merge_started;
+        Mutex.lock merge
+    | `After -> Mutex.lock merge_ended
```

Comment:
```suggestion
    | `After -> Alcotest.fail "Merge should have been aborted by clear"
```

I think it's better to fail than to block forever here (and we never unlock this `merge_ended` lock anyway.)